ci: pre compiled archive (linux)

pacien opened the pull request:
GitHub: closes #154

pacien commented:
This produces a somewhat portable pre-compiled version of the program for Linux AMD64.
It's not meant to be used, as the recommended way to install it would be through distro packages for Linux and MacOS.

@OzoneGrif I'll let you take care of the generation of such an archive for the Windows platform. You can take a look at the Travis CI build script and translate it for that platform, creating another build job running on a Windows machine.

OzoneGrif commented:
Please add some documentation about this commit.
> compiler: copy viewer from the exe's dir if it exists

pacien commented:
I added the possibility for the user to explicitly provide the path of the viewer to include when using `--with-viewer[=dist]` instead of defaulting to the executable's path, which was quite unpredictable.  I updated the manual page accordingly.

A relative path can be specified for this parameter, allowing the use of the viewer shipped in the archive.
